### PR TITLE
Add custom keymap 'j4ckofalltrades' for sofle/rev1

### PR DESCRIPTION
**Pull request by j4ckofalltrades**
## Description
    
Modified version of default Via-compatible keymap with focus on adding a standard navigation cluster layer
plus some Vim-inspired features e.g. soft escape (Esc when held, Ctrl when tapped), using 'h', 'j', 'k', 'l' as arrow keys.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sofle/keymaps/j4ckofalltrades/config.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2021 Jordan Duabe
```
Comment (drashna) [inline keyboards/sofle/keymaps/j4ckofalltrades/config.h:1]: This file can be removed.

File: keyboards/sofle/keymaps/j4ckofalltrades/encoder.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+#ifdef ENCODER_ENABLE
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
```
Comment (drashna) [inline keyboards/sofle/keymaps/j4ckofalltrades/encoder.c:24]:
```suggestion
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
```

File: keyboards/sofle/keymaps/j4ckofalltrades/keymap.c [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "oled.c"
+#include "encoder.c"
```
Comment (drashna) [inline keyboards/sofle/keymaps/j4ckofalltrades/keymap.c:19]: You should not include C files like this.  Only header files should be included. 
```suggestion
```

File: keyboards/sofle/keymaps/j4ckofalltrades/encoder.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef ENCODER_ENABLE
```
Comment (drashna) [inline keyboards/sofle/keymaps/j4ckofalltrades/encoder.c:17]:
```suggestion
#ifdef ENCODER_ENABLE
#    include QMK_KEYBOARD_H
```

File: keyboards/sofle/keymaps/j4ckofalltrades/oled.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef OLED_ENABLE
```
Comment (drashna) [inline keyboards/sofle/keymaps/j4ckofalltrades/oled.c:17]:
```suggestion
#ifdef OLED_ENABLE
#    include QMK_KEYBOARD_H
```

File: keyboards/sofle/keymaps/j4ckofalltrades/rules.mk [hunk 6]
```diff
@@ -0,0 +1,7 @@
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
+ENCODER_ENABLE = yes
+CONSOLE_ENABLE = no
+EXTRAKEY_ENABLE = yes
+VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/sofle/keymaps/j4ckofalltrades/rules.mk:7]:
```suggestion
LTO_ENABLE = yes

SRC += oled.c encoder.c
```

File: keyboards/sofle/keymaps/j4ckofalltrades/rules.mk [hunk 7]
```diff
@@ -0,0 +1,7 @@
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
+ENCODER_ENABLE = yes
```
Comment (drashna) [inline keyboards/sofle/keymaps/j4ckofalltrades/rules.mk:3]: These are defined at the keyboard level, so aren't technically needed. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by j4ckofalltrades** [comment]
Suggestions applied, ready for next round of review.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**